src/validation/validation-refinements.ts

```diff
@@ -260,28 +260,88 @@ export const validateBeaconMetadataReferences: SuperRefinement<{
   console.log(JSON.stringify(missingBeaconHelperDefaults, null, 2));
 };
 
-export const validateBeaconSetsReferences: SuperRefinement<{
-  apis: Apis;
+export const validateBeaconSetMetadataReferences: SuperRefinement<{
+  beaconSets: BeaconSets;
   explorer: Explorer;
-}> = ({ apis, explorer }, ctx) => {
-  Object.entries(explorer.beaconSets).forEach(([beaconSetId, beaconIds]) => {
-    beaconIds.forEach((beaconId) => {
-      // Check if /data/apis/<apiName>/beacons contains a file with the beaconId
-      if (
-        !Object.values(apis).some((api) =>
-          Object.values(api.beacons).some((beacon) => {
-            return beacon.beaconId === beaconId;
-          })
-        )
-      ) {
-        ctx.addIssue({
-          code: z.ZodIssueCode.custom,
-          message: `Referenced beacon ${beaconId} is not defined in /data/apis/<apiName>/beacons`,
-          path: ['explorer', 'beaconSets', beaconSetId],
-        });
-      }
+}> = ({ beaconSets, explorer }, ctx) => {
+  Object.entries(explorer.beaconSetMetadata).forEach(([beaconSetId, beaconSetMetadata]) => {
+    // Check if /data/beaconSets contains a file with the beaconSetId
+
+    const beaconSet = Object.values(beaconSets).find((beaconSet) => beaconSet.beaconSetId === beaconSetId);
+    if (!beaconSet) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `Referenced beaconSet ${beaconSetId} is not defined in /data/beaconSets`,
+        path: ['explorer', 'beaconSetMetadata'],
+      });
+
+      return;
+    }
+
+    // Find any missing chains - where the chain is specified in pricingCoverage but not in the beaconSet itself
+    const missingChains = Object.keys(beaconSet.chains).filter(
+      (chain) => !Object.keys(beaconSetMetadata.pricingCoverage).includes(chain)
+    );
+    missingChains.forEach((chain) => {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `Referenced pricing coverage chain ${chain} is not defined in the target beaconSet`,
```

Comment:
Also same [here](https://github.com/api3dao/operations/blob/05fe81834f1eb95c5cff72e457bc61312b32f2d4/src/validation/validation-refinements.ts#L206).

Reply:
I think you're right. The validation message and the comment are saying one thing and the actual validation is doing the validation in the opposite direction. I basically copy and pasted the validation from [beaconMetadata](https://github.com/api3dao/operations/blob/bec627-beaconsets-market-placeholder/src/validation/validation-refinements.ts#L199-L209) so I'll update both messages 👍🏻

Reply:
I ended up changing the validation instead of the text because there we are actually validating data in beaconSetMetadata file. I just wonder if we would also need to validate the reference both ways 🤔

Reply:
Although extra operational burden 😸 , your decision makes more sense in my opinion. If key-value pairs of pricingCoverage always gonna be same in future (test-pricing-set-free for testnets, chain name for mainnets), instead of both way validation, delegating normalization script to fill automatically `pricingCoverage` field can be an alternative.

Reply:
Then I'll just keep the `pricingCoverage.chain` exists in `beaconSet.chain` validation and wait for @aquarat to return to confirm adding the `pricingCoverage` to normalization script (I don't know if pricing coverage names are going to follow current convention in the future)